packages/evm/test/incentives/ERC20PeggedIncentive.t.sol

```diff
@@ -0,0 +1,244 @@
+
+        // Attempt to claim more than the limit in a single claim => revert
+        vm.expectRevert(AIncentive.NotClaimable.selector);
+        incentive.claim(address(this), _encodeBoostClaim(6 ether));
+    }
+
+    //////////////////////////////////////
+    // ERC20PeggedIncentive.preflight //
+    //////////////////////////////////////
+
+    function testPreflight() public {
+        bytes memory preflightPayload = incentive.preflight(
+            abi.encode(
+                ERC20PeggedIncentive.InitPayload({
+                    asset: address(mockAsset),
+                    peg: address(pegAsset),
+                    reward: 1 ether,
+                    limit: 5 ether,
+                    manager: address(this)
+                })
+            )
+        );
+
+        ABudget.Transfer memory transfer = abi.decode(preflightPayload, (ABudget.Transfer));
+        assertEq(transfer.asset, address(mockAsset));
+        assertEq(transfer.target, address(incentive));
+        assertEq(abi.decode(transfer.data, (ABudget.FungiblePayload)).amount, 5 ether);
+    }
+
+    ////////////////////////////////////
+    // ERC20PeggedIncentive.clawback //
+    ////////////////////////////////////
+
+    function testClawback() public {
+        // Initialize the ERC20PeggedIncentive
+        _initialize(address(mockAsset), address(pegAsset), 1 ether, 5 ether, address(this));
+
+        // Clawback some tokens
+        bytes memory reclaimPayload =
+            abi.encode(AIncentive.ClawbackPayload({target: address(this), data: abi.encode(2 ether)}));
+        incentive.clawback(reclaimPayload);
+
+        // Check the balance and limit
+        assertEq(mockAsset.balanceOf(address(this)), 2 ether);
+        assertEq(incentive.limit(), 3 ether);
+    }
+
+    function testClawback_Unauthorized() public {
+        // Initialize the ERC20PeggedIncentive
+        _initialize(address(mockAsset), address(pegAsset), 1 ether, 5 ether, address(this));
+
+        // Attempt to call clawback from an unauthorized address
+        address unauthorized = address(0x1234);
+        vm.prank(unauthorized);
+        bytes memory reclaimPayload =
+            abi.encode(AIncentive.ClawbackPayload({target: address(this), data: abi.encode(2 ether)}));
+        vm.expectRevert(BoostError.Unauthorized.selector);
+        incentive.clawback(reclaimPayload);
+    }
+
+    function testGetPeg() public {
+        // Initialize the ERC20PeggedIncentive
+        _initialize(address(mockAsset), address(pegAsset), 1 ether, 5 ether, address(this));
+
+        // Check the peg address
+        assertEq(incentive.getPeg(), address(pegAsset));
+    }
+
+    //////////////////////////////////////////////////
+    // ERC20PeggedIncentive.getComponentInterface //
+    //////////////////////////////////////////////////
+
+    function testGetComponentInterface() public view {
+        // Retrieve the component interface
+        console.logBytes4(incentive.getComponentInterface());
```

Comment:
should this log be removed?

Reply:
We use these in most tests so you can see the component interfaces in testing